packages/api-request-builder/src/create-service.js

```diff
@@ -123,6 +124,29 @@ export default function createService (
       setDefaultParams.call(this)
       return uri
     },
+
+    // Call this method to parse an object as params
+    parse (params: ServiceBuilderDefaultParams): Object {
```

Comment:
`this.params` is like a `const`. The reference can't be changed.

Also, the defaults would be missing. And this would overwrite everything that has already been set, so `service.page(3).parse({ perPage: 5 }).build()` would ignore `page`, but service.parse({ perPage: 5 }).page(3)` would be not (although we currently don't return the full API).

This would also not validate `params` at all.

The "ideal solution" outlined above allows arbitrary mixing of `parse` and the other requests.